docs/guides/latest/platforms/rook.md

```diff
@@ -0,0 +1,678 @@
+---
+title: Rook | Stash
+description: Using Stash with Rook Storage Service
+menu:
+  product_stash_0.8.3:
+    identifier: platforms-rook
+    name: Rook
+    parent: platforms
+    weight: 50
+product_name: stash
+menu_name: product_stash_0.8.3
+section_menu_id: guides
+---
+
+# Using Stash with Rook Storage Service
+
+This guide will show you how to use Stash to backup and restore volumes of a Kubernetes workload in [Rook](https://rook.io/) storage service. Here, we are going to backup a volume of a Deployment into [AWS S3](https://aws.amazon.com/s3/) compatible [Rook Object Storage](https://rook.io/docs/rook/v1.0/ceph-object.html). Then, we are going to show how to restore this volume into a PersistentVolumeClaim of [Rook Object Storage](https://rook.io/docs/rook/v1.0/ceph-object.html). We are going to also re-deploy deployment using this recovered volume.
+
+## Before You Begin
+
+- At first, you need to have a Kubernetes cluster. If you don't already have a cluster, you can create one by using [Minikube](https://github.com/kubernetes/minikube).
+
+- Install `Stash` in your cluster following the steps [here](https://appscode.com/products/stash/0.8.3/setup/install/).
+
+- You should be familiar with the following `Stash` concepts:
+  - [BackupConfiguration](/docs/concepts/crds/backupconfiguration.md/)
+  - [BackupSession](/docs/concepts/crds/backupsession.md/)
+  - [RestoreSession](/docs/concepts/crds/restoresession.md/)
+  - [Repository](/docs/concepts/crds/repository.md/)
+- You will need a [Rook Storage Service](https://rook.io/docs/rook/master/ceph-quickstart.html) with [Object Storage](https://rook.io/docs/rook/v1.0/ceph-object.html) and [Block Storage](https://rook.io/docs/rook/v1.0/ceph-block.html) configured. If you do not already have a Rook Storage Service configured, you can create one by following this [quickstart](https://rook.io/docs/rook/v1.0/ceph-quickstart.html) guide.
+
+To keep everything isolated, we are going to use a separate namespace called `demo` throughout this tutorial.
+
+```console
+$ kubectl create ns demo
+namespace/demo created
+```
+
+>**Note:** YAML files used in this tutorial are stored in  [docs/examples/guides/latest/platforms/rook](/docs/examples/guides/latest/platforms/rook) directory of [stashed/doc](https://github.com/stashed/doc) repository.
+
+**Ensure StorageClass:**
+
+[Rook Block Storage](https://rook.io/docs/rook/v1.0/ceph-block.html) allows mounting Rook storage into pod using a PersistentVolumeClaim. At first, we need to know respective [StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/) for Rook Block Storage by using the following command.
```

Comment:
[Rook Block Storage](https://rook.io/docs/rook/v1.0/ceph-block.html) allows mounting Rook storage into pod using a PersistentVolumeClaim. In order to do that, we have to create a PersistentVolumeClaim with `rook-ceph-block`[StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/). Verify the StorageClass exist by the following command: